cmd/loki-canary/main.go

```diff
@@ -83,6 +89,26 @@ func main() {
 		os.Exit(1)
 	}
 
+	var tlsConfig *tls.Config
+	tc := config.TLSConfig{}
+	if *certFile != "" || *keyFile != "" || *caFile != "" {
+		if !*useTLS {
+			_, _ = fmt.Fprintf(os.Stderr, "Must set --tls when specifying client certs\n")
+			os.Exit(1)
+		}
+		tc.CAFile = *caFile
+		tc.CertFile = *certFile
+		tc.KeyFile = *keyFile
+		tc.InsecureSkipVerify = false
+
+		var err error
+		tlsConfig, err = config.NewTLSConfig(&tc)
+		if err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "TLS configuration error: %s\n", err.Error())
+			os.Exit(1)
+		}
+	}
+
```

Comment:
Correct me if I'm missing anything. We could simplify this initialization code bit.

the gist is 
1. to first use `if useTLS ` to do all this and skip everything completely (even before checking for `*certFile, *keyFile, *caFile`.
2. And trusting `config.NewTLSConfig` for checking if all the files are invalid (even if its empty)

Something simple like
```go
	if useTLS {
		tls, err := config.NewTLSConfig(
			&config.TLSConfig{
				CAFile, *caFile,
				CertFile, *certFile,
				KeyFile, *keyFile,
				InsecureSkipVerify: false,
			},
		)
		if err != nil {
			// print error
			os.Exit(1)
		}
	}
}
```
No need to check for any empty string on those cert files. `NewTLSConfig()` should return error in that case correct?

Reply:
The main reason for doing this was to ensure valid configuration. Setting the client certs requires `-tls` to be set otherwise it is a noop. I actually made this mistake in my own testing and was glad to have the error message. On the other hand, when setting `-tls` it does not require setting client certs.